### PR TITLE
🧹`Marketplace`: Demonstrate usage of `decent_exposure`

### PR DESCRIPTION
**Pull request by zspencer**
This demonstrates how the `decent_exposure` gem reduces LoC-per-Controller. Right now, I've isolated this to the `Marketplace` namespace; since I wasn't sure if this would be received as a useful abstraction or an obstraction...

There's some warts I'm not keen on:

1. It uses `Object.const_get` to try and load a model based upon the name... which isn't the end of the world but it sure as heck doesn't play well with namespaces...
2. It would be nice if we could use a `with: [:policy_scoped]` so we don't have to call `policy_scope` every time; but it didn't look like that was possible straight-off-the-docs so I didn't go down the path of trying to make it work.

Thoughts?

P.s. The majority of the new lines are from the test I wrote to make sure that the `Marketplace::OrdersController` wasn't YOLOed. It's -38+18 if you take that into account, saving -6~7 lines-per-controller.

### CODE REVIEW

File: spec/furniture/marketplace/products_controller_request_spec.rb [hunk 1]
```diff
@@ -36,18 +36,19 @@
...
-      let(:product_attributes) { {} }
+      let(:product_attributes) { {description: "test"} }
 
       it { is_expected.to have_rendered(:new) }
     end
   end
 
   describe "#edit" do
     it "Shows the form to edit a Marketplace Product" do
+      sign_in(space, member)
```
Comment (zspencer) [inline spec/furniture/marketplace/products_controller_request_spec.rb:47]: !!! Apparently we were not authorizing the `edit` action until after we hit the view; which was allowing this test to "pass" even though no one was signed in...

Comment (anaulin) [reply spec/furniture/marketplace/products_controller_request_spec.rb:47]: Good catch.

**Review by anaulin — APPROVED** [review]
I like that this lets us dry up the whole "`find(id) || find(thing_params)`" side of things.

I don't like that it obscures how `authorize` is being called, and it is causing you to write some nonsensical-seeming code in this PR like:
```ruby
    def show
      order
    end
```
^ this seems to exist only to trigger the underlying fetching and authorization code, but that is not obvious and seems weird on first read. I'd much rather write `authorize order` here, or something that feels a bit more explicit or self-explanatory.

All in all I don't feel strongly either way. I'm curious what do others think.

File: app/furniture/marketplace/products_controller.rb [hunk 2]
```diff
@@ -2,11 +2,13 @@
 
 class Marketplace
   class ProductsController < Controller
+    expose :product, with: [:authorized], scope: -> { policy_scope(marketplace.products) }, model: Product
+
     def new
+      product
```
[the remark this answers is not in the document]
Comment (anaulin) [reply app/furniture/marketplace/products_controller.rb:8]: I apologize for the unnecessarily harsh word choice here, it's not called for. I had written this as a draft for myself and forgot to delete it before sending out the review. 🙏🏼

[the remark this answers is not in the document]
Comment (zspencer) [reply app/furniture/marketplace/products_controller.rb:8]: I mean it is useless nonsense; and is the part I disliked as well. I'm not sold on the `with: [:authorize]` bit

[the remark this answers is not in the document]
Comment (anaulin) [reply app/furniture/marketplace/products_controller.rb]: We could start with the regular `expose` bit, but authorize explicitly in the actions, see how we like it?

[the remark this answers is not in the document]
Comment (zspencer) [reply app/furniture/marketplace/products_controller.rb]: Yeah that's where I went but I need to figure out why things broke but I am officially done-with-computer for toda.

**Pull request merged**